Winter semester

Kobby-Adjei opened the pull request:
Pull Request: Semester Display Enhancement (#20)
This PR introduces a new feature that ensures instructors see the current semester and year upon logging into the Gradebook System.

Changes made:
Updated instructor.py to include the semester and year in the instructor's welcome message.

The get_current_semester() method determines the semester based on the current month.

The display_courses() method now prints the semester and year before listing available courses.

Added new tests in tests/test_main.py to verify semester display functionality using mocked dates.

Implemented @patch('instructor.datetime') to mock datetime.now() and return specific fixed dates:

May 1, 2000

January 1, 1999

October 15, 2024

Updated test_login_with_valid_id to validate that the correct semester and year are displayed when datetime.now() is mocked.

sjiang1 commented:
The branch is not up to date with main. Please merge the latest main to your branch.

sjiang1 commented:
pull request examples:
https://github.com/COSC481W-2024Fall/MLite/pull/17
https://github.com/COSC481W-2025Winter/SurvivalChess/pull/114

sjiang1 commented on tests/test_main.py:
```diff
@@ -46,6 +47,19 @@ def test_login_with_valid_id(monkeypatch, capsys, test_instructor):
     captured = capsys.readouterr()
+
+    # Determine expected semester and year
+    current_month = datetime.datetime.now().month
```
In the test, mock datetime.now() to return some arbitrary date instead of using the current date and time.

Kobby-Adjei commented:
**Pull Request Title**:  Fixed Tests After Merge Conflict Resolution & Synced with Latest Main (#20)

**Description:**

Resolved merge conflicts and successfully integrated changes from the latest main branch.

Updated test cases to ensure stability and accuracy.

Mocked datetime.now() in the test file to return a fixed date, preventing inconsistent results.

Verified functionality and checked for errors after merging.

sjiang1 commented on tests/test_main.py:
```diff
@@ -115,7 +122,7 @@ def test_select_valid_course(monkeypatch, capsys, test_instructor):
     # Cleanup
 
-def test_sort_courses(mocker, test_instructor):
+def test_sort_courses(mocker, test_instructor): # Added comma
```
I think this comment is unnecessary.

sjiang1 commented on tests/test_main.py:
```diff
@@ -77,6 +80,10 @@ def test_login_with_valid_id(monkeypatch, capsys, test_instructor):
     # Assert
     captured = capsys.readouterr()
+
+    # Since datetime is not mocked, we cannot assert the specific semester.
```
My previous comment
> In the test, mock datetime.now() to return some arbitrary date instead of using the current date and time.

is not addressed here.

Kobby-Adjei replied:
can you please clarify arbitary because i am assuming you want it to return a random date and not the current date and time

sjiang1 replied:
not the current date and time, for example, create a test where datetime.now() returns 5/1/2000
create a test where datetime.now() returns 1/1/1999, and create a test where datetime.now() returns 10/15/2024.

sjiang1 reviewed (changes requested):
I don't think my previous comment is addressed.

sjiang1 commented on tests/test_main.py:
```diff
@@ -66,9 +68,14 @@ def test_login_with_invalid_id(monkeypatch, capsys):
 # test if the user enters a valid digital id
-def test_login_with_valid_id(monkeypatch, capsys, test_instructor):
+@patch('instructor.datetime') # Mock the datetime module in instructor.py
+def test_login_with_valid_id(mock_dt_module, monkeypatch, capsys, test_instructor):
     # Arrange
-    responses = iter([test_instructor["id"], 'q'])
+    # Mock datetime.datetime.now() to return an arbitrary date
+    mock_dt_module.datetime.now.return_value = datetime.datetime(2024, 9, 15)
+    expected_semester_year_string = "Fall 2024"
```
Since all the test code use monkeypatch, is there any reason that you don't use monkeypatch for datetime?
If not, please revise all the test code so that you use monkeypatch for the mocking.

Kobby-Adjei replied:
### Fixed Tests to Use Monkeypatch for Datetime ( #20 )

sjiang1 reviewed (approved):
Good job!
